src/Symfony/Bundle/FrameworkBundle/EventListener/SuggestMissingPackageSubscriber.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+use Symfony\Component\Console\Exception\NamespaceNotFoundException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Suggests a package, that should be installed (via composer),
+ * if the package is missing, and the input command namespace can be mapped to a Symfony bundle.
+ *
+ * @author Przemysław Bogusz <przemyslaw.bogusz@tubotax.pl>
+ */
+final class SuggestMissingPackageSubscriber implements EventSubscriberInterface
+{
+    /**
+     * Mappings between namespaces of Symfony commands and packages required to run these commands.
+     *
+     * @var array
+     */
+    private static $suggestedPackages = [
+        'doctrine' => [
+            'Doctrine ORM',
+            'symfony/orm-pack',
+        ],
+        'generate' => [
+            'SensioGeneratorBundle',
+            'sensio/generator-bundle',
+        ],
+        'make' => [
+            'MakerBundle',
+            'symfony/maker-bundle --dev',
+        ],
+        'server' => [
+            'Symfony Web Server',
+            'symfony/web-server-bundle --dev',
+        ],
+    ];
+
+    public function onConsoleError(ConsoleErrorEvent $event)
+    {
+        if ($event->getError() instanceof NamespaceNotFoundException) {
```

Comment:
Please use one guard clause instead:

```php
if (!$event->getError() instanceof NamespaceNotFoundException || !$namespace = $event->getInput()->getFirstArgument() || !isset(self::SUGGESTED_PACKAGES[$namespace])) {
    return;
}

Reply:
I switched to guard clause but it's a little different since I modified the whole logic